joblib/test/test_numpy_pickle.py

```diff
@@ -245,15 +295,74 @@ def test_masked_array_persistence():
     nose.tools.assert_true(isinstance(b, np.ma.masked_array))
 
 
-def test_z_file():
-    # Test saving and loading data with Zfiles
+@with_numpy
+def test_compress_mmap_mode_warning():
+    # Test the warning in case of compress + mmap_mode
+    rnd = np.random.RandomState(0)
+    a = rnd.random_sample(10)
+    this_filename = env['filename'] + str(random.randint(0, 1000))
+    numpy_pickle.dump(a, this_filename, compress=1)
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
Bonus points if you rename `w` to something like `caught_warnings` or a better name.